cpu/fe310/include/context_frame.h

```diff
@@ -83,43 +83,42 @@ struct context_switch_frame {
  * @{
  */
 /* These values are checked for correctness in context_frame.c */
-#define pad_OFFSET    0
-#define pc_OFFSET     8
-#define s0_OFFSET     12
-#define s1_OFFSET     16
-#define s2_OFFSET     20
-#define s3_OFFSET     24
-#define s4_OFFSET     28
-#define s5_OFFSET     32
-#define s6_OFFSET     36
-#define s7_OFFSET     40
-#define s8_OFFSET     44
-#define s9_OFFSET     48
-#define s10_OFFSET    52
-#define s11_OFFSET    56
-#define ra_OFFSET     60
-#define tp_OFFSET     64
-#define t0_OFFSET     68
-#define t1_OFFSET     72
-#define t2_OFFSET     76
-#define t3_OFFSET     80
-#define t4_OFFSET     84
-#define t5_OFFSET     88
-#define t6_OFFSET     92
-#define a0_OFFSET     96
-#define a1_OFFSET     100
-#define a2_OFFSET     104
-#define a3_OFFSET     108
-#define a4_OFFSET     112
-#define a5_OFFSET     116
-#define a6_OFFSET     120
-#define a7_OFFSET     124
+#define s0_OFFSET     0
+#define s1_OFFSET     4
+#define s2_OFFSET     8
+#define s3_OFFSET     12
+#define s4_OFFSET     16
+#define s5_OFFSET     20
+#define s6_OFFSET     24
+#define s7_OFFSET     28
+#define s8_OFFSET     32
+#define s9_OFFSET     36
+#define s10_OFFSET    40
+#define s11_OFFSET    44
+#define ra_OFFSET     48
+#define t0_OFFSET     52
+#define t1_OFFSET     56
+#define t2_OFFSET     60
+#define t3_OFFSET     64
+#define t4_OFFSET     68
+#define t5_OFFSET     72
+#define t6_OFFSET     76
+#define a0_OFFSET     80
+#define a1_OFFSET     84
+#define a2_OFFSET     88
+#define a3_OFFSET     92
+#define a4_OFFSET     96
+#define a5_OFFSET     100
+#define a6_OFFSET     104
+#define a7_OFFSET     108
+#define pc_OFFSET     112
+#define pad_OFFSET    116
 /** @} */
 
 /**
  * @brief Size of context switch frame
  */
-#define CONTEXT_FRAME_SIZE (a7_OFFSET + 4)
+#define CONTEXT_FRAME_SIZE (pad_OFFSET + 12)
```

Comment:
Why 12?

Reply:
According to the RISC-V ABI, the stack pointer must be advanced a multiple of 16 bytes. I removed the saving/restoring of the thread pointer (tp) register because I was altering that register anyway as part of the context switching interrupt handling. The padding for the context frame structure needs 12 additional bytes at the end to be a multiple of 16 (28 registers saved * 4bytes + 12bytes = 128bytes for the stack frame). The padding was at the beginning of the structure - I moved it to the end, thus the +12.

Reply:
So you want to do `sizeof(struct context_switch_frame)`

Reply:
No because `CONTEXT_FRAME_SIZE` needs to be used in the assembly code interrupt trap handler in intr.S file. The `sizeof()` won't be recognized by the assembler.

Reply:
bummer, but I see you have `_Static_assert`s to check that the defines match the struct - so all is well :)